cpp/tests/utilities/test_graphs.hpp

```diff
@@ -212,79 +225,112 @@ class Rmat_Usecase : public detail::TranslateGraph_Usecase {
       }
 
-      if (i > 0) {
-        auto start_offset = src_v.size();
-        src_v.resize(start_offset + tmp_src_v.size(), handle.get_stream());
-        dst_v.resize(start_offset + tmp_dst_v.size(), handle.get_stream());
-        raft::copy(
-          src_v.begin() + start_offset, tmp_src_v.begin(), tmp_src_v.size(), handle.get_stream());
-        raft::copy(
-          dst_v.begin() + start_offset, tmp_dst_v.begin(), tmp_dst_v.size(), handle.get_stream());
-
-        if (weights_v) {
-          weights_v->resize(start_offset + tmp_weights_v->size(), handle.get_stream());
-          raft::copy(weights_v->begin() + start_offset,
-                     tmp_weights_v->begin(),
-                     tmp_weights_v->size(),
-                     handle.get_stream());
-        }
+      translate(handle, tmp_src_v, tmp_dst_v);
+
+      if (undirected_) {
+        std::tie(tmp_src_v, tmp_dst_v, tmp_weights_v) =
+          cugraph::symmetrize_edgelist_from_triangular<vertex_t, weight_t>(
+            handle, std::move(tmp_src_v), std::move(tmp_dst_v), std::move(tmp_weights_v));
       }
-    }
 
-    translate(handle, src_v, dst_v);
+      if (multi_gpu) {
+        std::tie(store_transposed ? tmp_dst_v : tmp_src_v,
+                 store_transposed ? tmp_src_v : tmp_dst_v,
+                 tmp_weights_v) =
+          cugraph::detail::shuffle_edgelist_by_gpu_id(
+            handle,
+            store_transposed ? std::move(tmp_dst_v) : std::move(tmp_src_v),
+            store_transposed ? std::move(tmp_src_v) : std::move(tmp_dst_v),
+            std::move(tmp_weights_v));
+      }
 
-    if (undirected_)
-      std::tie(src_v, dst_v, weights_v) =
-        cugraph::symmetrize_edgelist_from_triangular<vertex_t, weight_t>(
-          handle, std::move(src_v), std::move(dst_v), std::move(weights_v));
+      src_partitions.push_back(std::move(tmp_src_v));
+      dst_partitions.push_back(std::move(tmp_dst_v));
+      if (weight_partitions) { (*weight_partitions).push_back(std::move(*tmp_weights_v)); }
+    }
 
-    if (multi_gpu) {
-      std::tie(store_transposed ? dst_v : src_v, store_transposed ? src_v : dst_v, weights_v) =
-        cugraph::detail::shuffle_edgelist_by_gpu_id(
-          handle,
-          store_transposed ? std::move(dst_v) : std::move(src_v),
-          store_transposed ? std::move(src_v) : std::move(dst_v),
-          std::move(weights_v));
+    size_t tot_edge_counts{0};
+    for (size_t i = 0; i < src_partitions.size(); ++i) {
+      tot_edge_counts += src_partitions[i].size();
     }
 
-    rmm::device_uvector<vertex_t> vertices_v(0, handle.get_stream());
-    for (size_t i = 0; i < partition_ids.size(); ++i) {
-      auto id = partition_ids[i];
+    rmm::device_uvector<vertex_t> src_v(tot_edge_counts, handle.get_stream());
+    size_t src_offset{0};
+    for (size_t i = 0; i < src_partitions.size(); ++i) {
+      thrust::copy(handle.get_thrust_policy(),
```

Comment:
Yeah... I wonder why compilation didn't fail. Maybe thrust copy is not using any `__deivce__` functions (just calling cudaMempyAsync inside), but anyways, I will switch to raft::copy to avoid using thrust functions in .hpp file.